### PR TITLE
Add render_parent helper method to avoid double-rendering hijinks

### PR DESCRIPTION
**Pull request by camertron**
### Summary

Although perhaps not a common pattern, it is possible to call `super` in a derived component to render the parent's template:

```html+erb
<%# base_component.html.erb %>
<div>Content from base component</div>

<%# derived_component.html.erb %>
<div>Content from derived component</div>
<% super %>
```

```ruby
# base_component.rb
class BaseComponent < ViewComponent::Base
end

# derived_component.rb
class DerivedComponent < BaseComponent
end
```

Rendering `DerivedComponent` results in the following, as expected:

```html
<div>Content from derived component</div>
<div>Content from base component</div>
```

Unfortunately however it is very tempting to echo the result of `super` out to the page, eg. `<%= super %>` instead of `<% super %>`, which leads to double rendering both components.

This PR introduces a new `render_parent` method that simply calls `super` and returns `nil`.

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,12 @@ title: Changelog
 
 ## main
 
+## 2.55.0
+
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:13]:
```suggestion
```

These are added by the release script ❤️

File: docs/guide/templates.md [hunk 2]
```diff
@@ -92,3 +92,14 @@ Component subclasses inherit the parent component's template if they don't defin
 class MyLinkComponent < LinkComponent
 end
 ```
+
+### Rendering the Parent's Template
```
Comment (joelhawksley) [inline docs/guide/templates.md:96]:
```suggestion
### Rendering parent templates
```

File: docs/guide/templates.md [hunk 3]
```diff
@@ -92,3 +92,14 @@ Component subclasses inherit the parent component's template if they don't defin
...
+
+Subclasses that wish to render the parent's template may do so by calling the `render_parent` helper method.
```
Comment (joelhawksley) [inline docs/guide/templates.md:98]:
```suggestion
To render a parent component's template from a subclass, call `render_parent`:
```

File: docs/guide/templates.md [hunk 4]
```diff
@@ -92,3 +92,14 @@ Component subclasses inherit the parent component's template if they don't defin
...
+
+```ruby
```
Comment (joelhawksley) [inline docs/guide/templates.md:100]: This should be:

```suggestion
```erb
```

right?

File: lib/view_component/base.rb [hunk 5]
```diff
@@ -117,6 +117,24 @@ def perform_render
       render_template_for(@__vc_variant).to_s + _output_postamble
     end
 
+    # Superclass components that call `super` inside their template code will cause a
```
Comment (joelhawksley) [inline lib/view_component/base.rb:120]:
```suggestion
    # Subclass components that call `super` inside their template code will cause a
```

File: lib/view_component/base.rb [hunk 6]
```diff
@@ -117,6 +117,24 @@ def perform_render
...
+    # double render if they accidentally emit the result. In other words, this double
+    # renders:
```
Comment (joelhawksley) [inline lib/view_component/base.rb:122]:
```suggestion
    # double render if they accidentally emit the result:
```

File: lib/view_component/base.rb [hunk 7]
```diff
@@ -117,6 +117,24 @@ def perform_render
...
+    #
+    # <%= super %>
```
Comment (joelhawksley) [inline lib/view_component/base.rb:124]:
```suggestion
    # <%= super %> # double-renders
```

File: lib/view_component/base.rb [hunk 8]
```diff
@@ -117,6 +117,24 @@ def perform_render
...
+    #
+    # but this does not:
+    #
+    # <% super %>
+    #
```
Comment (joelhawksley) [inline lib/view_component/base.rb:129]:
```suggestion
    # <% super %> # does not double-render
    #
```

File: lib/view_component/base.rb [hunk 9]
```diff
@@ -117,6 +117,24 @@ def perform_render
...
+    # The render_parent method simply calls `super` but makes sure to return nil to
+    # avoid rendering the result twice.
```
Comment (joelhawksley) [inline lib/view_component/base.rb:131]:
```suggestion
    # Calls `super`, returning nil to avoid rendering the result twice.
```

**Review by joelhawksley — APPROVED** [review]

File: docs/guide/templates.md (hunk 4 above)
Comment (camertron) [reply docs/guide/templates.md:100]: Woops, yes, thanks.

**Pull request merged**